torch/ao/quantization/fx/quantization_patterns.py

```diff
@@ -955,15 +956,23 @@ def convert(self,
                 # 2. select corresponding quantized linear class for the float linear class
                 if activation_int8_quantized:
                     additional_static_quant_mapping = convert_custom_config_dict.get("static", {})
-                    qlinear = get_static_quant_module_class(
-                        type(self.linear), additional_static_quant_mapping)
+                    if module_has_sparse_params(self.linear):
+                        mapping_fn = get_static_sparse_quant_module_class
+                    else:
+                        mapping_fn = get_static_quant_module_class
+                    qlinear = mapping_fn(type(self.linear),
+                                         additional_static_quant_mapping)
```

Comment:
this still means that quantization knows about sparsity.  If we want them to be fully independent, we could move this to an external API.  Something like

```
prepare_custom_config_dict = {
  'float_to_observed_custom_module_class_with_lambda': {
    'static': {
      'float_class': torch.nn.Linear,
      'observed_class': torch.ao.nn.sparse.Linear,  # or whatever the type is
      'lambda': lambda x: module_has_sparse_params(x),
    },
  },
  # repeat for observed to quantized
}
```

and then make the quantization prepare and convert honor these swaps with lambda

or something similar